internal/jujuapi/interface.go

```diff
@@ -3,55 +3,23 @@
-	DestroyOffer(ctx context.Context, user *openfga.User, offerURL string, force bool) error
-	FindApplicationOffers(ctx context.Context, user *openfga.User, filters ...jujuparams.OfferFilter) ([]jujuparams.ApplicationOfferAdminDetailsV5, error)
-	ForEachCloud(ctx context.Context, user *openfga.User, f func(*dbmodel.Cloud) error) error
-	ForEachUserCloud(ctx context.Context, user *openfga.User, f func(*dbmodel.Cloud) error) error
-	ForEachUserCloudCredential(ctx context.Context, u *dbmodel.Identity, ct names.CloudTag, f func(cred *dbmodel.CloudCredential) error) error
-	GetApplicationOffer(ctx context.Context, user *openfga.User, offerURL string) (*jujuparams.ApplicationOfferAdminDetailsV5, error)
-	GetApplicationOfferConsumeDetails(ctx context.Context, user *openfga.User, details *jujuparams.ConsumeOfferDetails, v bakery.Version) error
-	GetCloud(ctx context.Context, u *openfga.User, tag names.CloudTag) (dbmodel.Cloud, error)
-	GetCloudCredential(ctx context.Context, user *openfga.User, tag names.CloudCredentialTag) (*dbmodel.CloudCredential, error)
-	GetCloudCredentialAttributes(ctx context.Context, u *openfga.User, cred *dbmodel.CloudCredential, hidden bool) (attrs map[string]string, redacted []string, err error)
 	RoleManager() jimm.RoleManager
 	GroupManager() jimm.GroupManager
 	IdentityManager() jimm.IdentityManager
 	LoginManager() jimm.LoginManager
 	PermissionManager() jimm.PermissionManager
 	AuditLogManager() jimm.AuditLogManager
 	ServiceAccountManager() jimm.ServiceAccountManager
+	JujuManager() jimm.JujuManager
 
-	InitiateInternalMigration(ctx context.Context, user *openfga.User, modelNameOrUUID string, targetController string) (jujuparams.InitiateMigrationResult, error)
-	InitiateMigration(ctx context.Context, user *openfga.User, spec jujuparams.MigrationSpec) (jujuparams.InitiateMigrationResult, error)
-	ListApplicationOffers(ctx context.Context, user *openfga.User, filters ...jujuparams.OfferFilter) ([]jujuparams.ApplicationOfferAdminDetailsV5, error)
-	ListModels(ctx context.Context, user *openfga.User) ([]base.UserModel, error)
-	Offer(ctx context.Context, user *openfga.User, offer jimm.AddApplicationOfferParams) error
-	PubSubHub() *pubsub.Hub
-	RemoveCloud(ctx context.Context, u *openfga.User, ct names.CloudTag) error
-	RemoveCloudFromController(ctx context.Context, u *openfga.User, controllerName string, ct names.CloudTag) error
-	RemoveController(ctx context.Context, user *openfga.User, controllerName string, force bool) error
 	ResourceTag() names.ControllerTag
-	RevokeCloudCredential(ctx context.Context, user *dbmodel.Identity, tag names.CloudCredentialTag, force bool) error
-
-	UpdateCloud(ctx context.Context, u *openfga.User, ct names.CloudTag, cloud jujuparams.Cloud) error
-	UpdateCloudCredential(ctx context.Context, u *openfga.User, args jimm.UpdateCloudCredentialArgs) ([]jujuparams.UpdateCredentialModelResult, error)
+	PubSubHub() *pubsub.Hub
```

Comment:
This method is still here and it leaks a type from our business layer into the API layer. The API layer uses it to create a watcher. In a future PR I would like the business layer to return a watcher directly rather than giving our pubSubHubs.

Reply:
yes, i think we can remove the model summary watcher..